wagtail/wagtailcore/fields.py

```diff
@@ -35,8 +45,13 @@ def value_from_datadict(self, data, files, name):
 
 
 class RichTextField(models.TextField):
+
+    def __init__(self, hallo_config=None, **kwargs):
```

Comment:
I highly agree with the idea of passing a custom widget instead of a type specific configuration into either the RichTextEditor or RichTextBlock. It feels a lot more consistent to the general django forms workflow.

On behalf of the naming convention, I would agree more with Josh. Calling that attribute plugin, which passes customising configuration information, might be a bit confusing. Especially when planning for future versatility towards possible other editor integration.

I will update my code to reflect these discussed changes.